kernel/src/drivers/terminal/mod.rs

```diff
@@ -0,0 +1,316 @@
+use core::ops::Add;
+use core::fmt;
+use core::result::Result;
+use spin::Mutex;
+use drivers::vga::{self, VgaWriter};
+use color::Color;
+
+pub static WRITER: Mutex<VgaWriter> = Mutex::new(VgaWriter::new());
+pub static STDOUT: Mutex<TextArea<VgaWriter>> = Mutex::new(TextArea::new(&WRITER, Point::new(0, 0), Point::new(vga::RESOLUTION_X, vga::RESOLUTION_Y)));
```

Comment:
Why is stdout a text area?

Reply:
That's what it is. It's no different from a text area.

Reply:
It takes up the whole of the output though? Shouldn't it hence be a Terminal?